lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -475,26 +473,30 @@ def detach_eni(eni, module):
         module.exit_json(changed=False, interface=get_eni_info(eni))
 
 
-def find_eni(connection, module):
+def uniquely_find_eni(connection, module):
 
     eni_id = module.params.get("eni_id")
-    subnet_id = module.params.get('subnet_id')
     private_ip_address = module.params.get('private_ip_address')
     instance_id = module.params.get('instance_id')
     device_index = module.params.get('device_index')
 
     try:
         filters = {}
-        if subnet_id:
-            filters['subnet-id'] = subnet_id
+
+        # proceed only if we're univocally specifying an ENI
+        if eni_id is None and private_ip_address is None and (instance_id is None and device_index is None):
+            return None
+
         if private_ip_address:
             filters['private-ip-address'] = private_ip_address
         else:
-            if instance_id:
+            if instance_id and device_index:
                 filters['attachment.instance-id'] = instance_id
-            if device_index:
                 filters['attachment.device-index'] = device_index
 
+        if eni_id is None and len(filters) == 0:
```

Comment:
Oh, nevermind, you're right about this. I guess the filters are entirely optional and it will just get all of the enis in this case.